File: cis/lib/fetchdata/bookingruangan.dart
class BookingRuangan {
  final int id;
  final int mahasiswaId;
  final int ruanganId;
  final String rencanaMulai;
  final String rencanaBerakhir;
  final String keperluan;
  final int status;

  BookingRuangan({
    required this.id,
    required this.mahasiswaId,
    required this.ruanganId,
    required this.rencanaMulai,
    required this.rencanaBerakhir,
    required this.keperluan,
    required this.status,
  });

  factory BookingRuangan.fromJson(Map<String, dynamic> json) {
    return BookingRuangan(
      id: json['id'],
      mahasiswaId: json['mahasiswa_id'],
      ruanganId: json['ruangan_id'],
      rencanaMulai: json['RencanaMulai'],
      rencanaBerakhir: json['RencanaBerakhir'],
      keperluan: json['Keperluan'],
      status: json['status'],
    );
  }
}

File: web-service/app/Models/BookingRuangan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Notifications\Notifiable;

class BookingRuangan extends Model
{
    use HasFactory, Notifiable;
    protected $fillable = [
        'mahasiswa_id',
        'ruangan_id',
        'RencanaMulai',
        'RencanaBerakhir',
        'Keperluan',
        'status',
    ];

    public function Mahasiswa()
    {
        return $this->belongsTo(Mahasiswa::class, 'mahasiswa_id');
    }

    public function Ruangan()
    {
        return $this->belongsTo(Ruangan::class, 'ruangan_id');
    }
}
